src/codeu/chat/client/core/ConversationContext.java

```diff
@@ -66,6 +66,19 @@ public MessageContext firstMessage() {
         null :
         getMessage(updated.firstMessage);
   }
+  
+  public int getMessageCount() {
+	  MessageContext currentMessage = firstMessage();
+	  if (currentMessage == null) {
+		  return 0;
+	  }
+	  int count = 1; // start with first message
+	  while(currentMessage != null && currentMessage.next() != null) {
```

Comment:
What if the conversation contains a million messages? Would you still loop through all of them every time?

(Not saying you should change anything right now. But this is Google's favorite question, so I figured I'd try to get your brain thinking about how stuff would work at scale.)